content/open-source/concepts/node-js/node-js.md

```diff
@@ -0,0 +1,165 @@
+
+1. [Download and install](https://nodejs.org/en/download/) Node.js for a specific operating system.
+2. To run JS files in Node, the `node` command followed by a [file path](https://www.codecademy.com/resources/docs/general/file-paths) will execute the program file.
+
+For example, if we have the following saved in a `script.js` file:
+
+```js
+console.log('I am a Node program');
+```
+
+We can run the terminal command in the same folder as `script.js`:
+
+```shell
+node script.js
+```
+
+This will start Node, print `I am a Node program` to the terminal window, and then exit, as the script file has finished execution.
+
+## Node as a REPL
+
+Node can also be used in a terminal as a read-evaluate-print-loop (REPL). This functionality allows JS commands to be executed from the terminal.
+
+With Node installed, the REPL can be launched with the `node` command in a terminal and pressing <kbd>Enter</kbd>. This launches the interactive JS environment where any valid JS code (e.g., `4 + 5`) can be run. After executing a command, Node will always print the result of that evaluation:
+
+```shell
+$ node
+> 4 + 5
+9
+> function nodeIsGreat() {
+... console.log('Node is great!');
+... }
+undefined
+> nodeIsGreat()
+Node is great!
+undefined
+> .exit
+$
+```
+
+In this example, the user launches node on line one with the `node` terminal command.
+
+On line 2 the user types `4 + 5` and evaluates with the <kbd>Enter</kbd> key. `9` prints to the output terminal.
+
+On line 4, the user opens a function declaration of `nodeIsGreat()`. Because this function declaration takes multiple lines, Node REPL will print `...` at the beginning of a line to show that it is still reading the user's input statement and has not evaluated yet. After the function declaration is closed on line 6, `undefined` prints to the output terminal, as the function declaration itself does not evaluate to any value.
+
+When the function is invoked on line 8, `Node is great!` logs to the console, and `undefined` logs after, because `nodeIsGreat()` returns `undefined`.
+
+To exit the Node REPL, use the `.exit` command at any point and return to the system shell. Pressing <kbd>Ctrl</kbd> + <kbd>C</kbd> twice will also exit.
+
+### Loading Existing Files
+
+Node REPL can also load existing JS files. If we had the following code saved into `script.js`:
+
+```js
+var a = 'Node REPL is fun!';
+```
+
+We can use `.load` to load it into the REPL. `.load` takes a path argument, so to load `script.js` we would use `.load ./script.js`.
+
+```shell
+$ node
+> .load ./script.js
+var a = 'Node REPL is fun!';
+
+> a
+'Node REPL is fun!'
+```
+
+After the script file is loaded, the variables are accessible in the REPL, so when we evaluate the `a` variable, it's value has been set by loading `script.js`, and 'Node REPL is fun!' prints to the console.
+
+This can be tested by running `node` in a terminal or check out the <a href="https://nodejs.org/api/repl.html" >REPL docs</a> for more functionality.
```

Comment:
```suggestion
This can be tested by running `node` in a terminal or check out the <a href="https://nodejs.org/api/repl.html" >REPL docs</a> for more information on commands and functionality.
```